app/models/source.rb

```diff
@@ -12,13 +12,13 @@ def default_endpoint
 
 
   # Container Inventory Objects
-  has_many :container_groups,    :dependent => :destroy
-  has_many :container_templates, :dependent => :destroy
-  has_many :container_projects,  :dependent => :destroy
-  has_many :container_nodes,     :dependent => :destroy
+  has_many :container_groups,    -> { active }, :dependent => :destroy
```

Comment:
so, if everything is targeted, this might not be needed?
So instead, this would be an API filter? Or I am trying to think about how will we query all these.

Reply:
Without this we always pick up records that were already deleted when doing the save:

e.g.: `D, [2018-10-15T13:51:58.266408 #19477] DEBUG -- : Processing InventoryCollection:<ContainerGroup>, created=0, updated=102, deleted=46...Complete`
These 46 were already deleted

Reply:
that shouldn't be happening if the :targeted is true